tests/integration/rosetta/rosetta_test.go

```diff
@@ -0,0 +1,253 @@
+var rosettaSvc *rosetta.Data
+
+func TestMain(m *testing.M) {
+
+	const testDbBackupFile = "rosetta-get-balance-test.db"
+	opts := badger.DefaultOptions("").
+		WithInMemory(true).
+		WithReadOnly(true).
+		WithLogger(nil)
+
+	db, err := badger.Open(opts)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	dbSnapshot, err := os.Open(testDbBackupFile)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	defer dbSnapshot.Close()
+
+	err = db.Load(dbSnapshot, 10)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	core, err := state.NewCoreFromDB(db)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	// setup scaffolding for minimal server we need for the tests
+	chain := flow.ChainID("flow-testnet").Chain()
+	validate := validator.New(chain, core.Height())
+	headers := invoker.NewHeaders(core.Chain())
+	invoke := invoker.New(zerolog.Nop(), core, chain, headers)
+	retrieve := retriever.New(invoke)
+
+	rosettaSvc = rosetta.NewData(validate, retrieve)
+
+	os.Exit(m.Run())
+}
+
+func TestGetBalance(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		wantStatusCode int
+		request        rosetta.BalanceRequest
+		wantBalance    string
+		wantHandlerErr assert.ErrorAssertionFunc
+	}{
+		// TODO: use a different 'valid balance request' - one that has an actual, non-zero balance
+		{
+			name: "valid balance request",
+			request: rosetta.BalanceRequest{
+				NetworkID: getDefaultNetworkID(),
+				AccountID: identifier.Account{
+					Address: "8c5303eaa26202d6",
+				},
+				BlockID: identifier.Block{
+					Index: 0,
+					Hash:  "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333",
+				},
+				Currencies: getDefaultCurrencySpec(),
+			},
+			wantBalance:    "0",
+			wantStatusCode: http.StatusOK,
+			wantHandlerErr: assert.NoError,
+		},
+		{
+			name:           "empty balance request",
+			request:        rosetta.BalanceRequest{},
+			wantStatusCode: http.StatusUnprocessableEntity,
+			wantHandlerErr: assert.Error,
+		},
+		{
+			name: "block hash and height mismatch",
+			request: rosetta.BalanceRequest{
+				NetworkID: getDefaultNetworkID(),
+				AccountID: identifier.Account{
+					Address: "8c5303eaa26202d6",
+				},
+				BlockID: identifier.Block{
+					Index: 99,
+					Hash:  "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333",
+				},
+				Currencies: getDefaultCurrencySpec(),
+			},
+			wantStatusCode: http.StatusUnprocessableEntity,
+			wantHandlerErr: assert.Error,
+		},
+		{
+			name: "invalid account address",
+			request: rosetta.BalanceRequest{
+				NetworkID: getDefaultNetworkID(),
+				AccountID: identifier.Account{
+					Address: "invalid_address",
+				},
+				BlockID: identifier.Block{
+					Index: 0,
+					Hash:  "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333",
+				},
+				Currencies: getDefaultCurrencySpec(),
+			},
+			wantStatusCode: http.StatusUnprocessableEntity,
+			wantHandlerErr: assert.Error,
+		},
+	}
+
+	for _, test := range tests {
+
+		t.Run(test.name, func(t *testing.T) {
+
```

Comment:
Is it possible to run those tests in parallel or do they need to be sequential? If it's possible to parallelize them, I'd suggest

```suggestion
		test := test
		t.Run(test.name, func(t *testing.T) {
			t.Parallel()
```